js/ui/list/ui.list.base.js

```diff
@@ -531,7 +533,21 @@ export const ListBase = CollectionWidget.inherit({
 
         if(isElementVisible && !this._scrollViewIsFull() && !this._isDataSourceLoading() && !this._isLastPage()) {
             clearTimeout(this._loadNextPageTimer);
-            this._loadNextPageTimer = setTimeout(this._loadNextPage.bind(this));
+            this._loadNextPageTimer = setTimeout(() => {
+                this._loadNextPage().done(this._setPreviousPageIfNewIsEmpty.bind(this));
+            });
+        }
+    },
+
+    _setPreviousPageIfNewIsEmpty: function(result) {
+        if(this.option('_revertPageOnEmptyLoad')) {
+            const dataSource = this._dataSource;
+            const pageIndex = dataSource?.pageIndex();
+
+            if(result && result.length === 0 && pageIndex > 0) {
+                this._fireContentReadyAction();
```

Comment:
Do we have to do this before changing the pageIndex?

Reply:
Yes, in other ways we can make infinite loops